### ISSUE TITLE
Unexpected default equivalency behaviour with interfaces

### ISSUE THREAD

**Issue by nullpainter** [fluentassertions/fluentassertions#2615]
### Description

When I am comparing two records or classes which contain a field that is defined as an interface, `NotBeEquivalentTo` and `BeEquivalentTo` ignore members which aren't defined in the interface. 

### Reproduction Steps

```csharp

[TestFixture]
public class EquivalencyTest
{
    [Test]
    public void TestEquivalency()
    {
        var first = new TestParent(new TestClass { MyField = "biscuits" });
        var second = new TestParent(new TestClass { MyField = "donkeys" });

        /*
Fails with:         
         
Expected first not to be equivalent to TestParent
{
    Something = TestClass
    {
        MyField = "donkeys"
    }
}, but they are.
         */
        first.Should().NotBeEquivalentTo(second);
    }
}

public class TestClass : ISomething
{
    public required string MyField { get; init; }
}

public record TestParent(ISomething Something);

public interface ISomething
{
}
```

### Expected behavior

Equivalency should fail as `MyField` is different between the two objects. The default behaviour is confusing and dangerous, as it can result in tests passing when they shouldn't. 

I'm sure there are good reasons for the architecture - and I now see the documentation which states that comparison is by default done against compile-time types - but I anticipated that these equivalency methods compare the _values_ in the object graph, so as an author of a test I shouldn't need to be concerned whether fields were defined as interfaces or not. From a test perspective, comparing models, this almost feels like an implementation detail.

As far as the test reads, I have two objects where the fields are patently different, but - with default config - Fluent Assertions doesn't think so. 

### Actual behavior

Equivalency passes when fields are defined as interfaces and properties being compared aren't in the interface.

### Known Workarounds

Use ` o => o.RespectingRuntimeTypes())` option or use `AssertionOptions.AssertEquivalencyUsing(o => o.RespectingRuntimeTypes());`

### Configuration

* FluentAssertions 6.12.0
* .NET 8.0

### Are you willing to help with a pull-request?

No

**Comment by nullpainter** [comment]
Actually, the following also fails. I've been using `ShouldBeEquivalentTo` for years now - I'm sure I would have noticed this by now, but perhaps not?

```csharp
        ISomething first = new TestClass { MyField = "biscuits" };
        ISomething second = new TestClass { MyField = "donkeys" };

        /*
Fails with:

No members were found for comparison. Please specify some members to include in the comparison or choose a more meaningful assertion.
         */
        first.Should().NotBeEquivalentTo(second);

**Comment by nullpainter** [comment]
I just noticed the `o => o.RespectingRuntimeTypes());` option in another ticket - shouldn't this be a default?

**Comment by dennisdoomen** [comment]
As you've concluded yourself, we had several discussions in the past and decided that the compile-time type is what most users are expecting. Considering the fact that you are the first to bring this up, confirms that assumption. 

Although I understand your issue, we have no plans to change the default behavior.

**Comment by nullpainter** [comment]
> Considering the fact that you are the first to bring this up, confirms that assumption.

Not necessary. I've used FluentValidation heavily since 2014 and this is the first time I've encountered this default behaviour - and that's only because I had changed something in a model which I expected to break in a test, but didn't. 

I appreciate that changing a default like this would potentially have a very large impact on users. This issue was initially raised as a bug as the default behaviour seemed - and still seems - non-obvious unless one reads the documentation. I am thankful that there is a global override though.
